packages/fxa-auth-server/lib/routes/utils/cms/localization.ts

```diff
@@ -738,4 +657,126 @@ export class CMSLocalization {
       throw error;
     }
   }
+
+  /**
+   * Fetch localized FTL content with fallback logic
+   * Uses the cached getFtlContent method which handles caching automatically
+   */
+  public async fetchLocalizedFtlWithFallback(locale: string): Promise<string> {
+    let hadErrors = false;
+
+    // First try the specific locale (e.g., 'en-US') using the cached getFtlContent method
+    try {
+      const ftlContent = await this.cmsManager.getFtlContent(locale, this.config);
+      // Even if content is empty, this is a successful fetch
+      this.statsd.increment('cms.getLocalizedConfig.ftl.success');
+      if (ftlContent) {
+        return ftlContent;
+      }
+    } catch (error) {
+      hadErrors = true;
+      // Log the failure but don't increment metrics yet
+      this.log.warn('cms.getLocalizedConfig.locale.failed', {
+        locale,
+        error: error.message
+      });
+    }
+
+    // If specific locale failed or returned empty content, try base language fallback
+    const baseLocale = this.extractBaseLocale(locale);
+    if (baseLocale && baseLocale !== locale) {
+      try {
+        this.log.info('cms.getLocalizedConfig.locale.fallback', {
+          originalLocale: locale,
+          fallbackLocale: baseLocale
+        });
+
+        const fallbackContent = await this.cmsManager.getFtlContent(baseLocale, this.config);
+        // Even if content is empty, this is a successful fetch
+        this.statsd.increment('cms.getLocalizedConfig.ftl.success');
+        if (fallbackContent) {
+          return fallbackContent;
+        }
+      } catch (error) {
+        hadErrors = true;
+        this.log.warn('cms.getLocalizedConfig.locale.fallback.failed', {
+          originalLocale: locale,
+          fallbackLocale: baseLocale,
+          error: error.message
+        });
+      }
+    }
+
+    // Both attempts failed or returned empty content
+    // Only increment ftl.fallback if we had actual errors, not just empty content
+    if (hadErrors) {
+      this.statsd.increment('cms.getLocalizedConfig.ftl.fallback');
+    }
+    return '';
+  }
+
+  /**
+   * Extract base locale from a locale string
+   */
+  public extractBaseLocale(locale: string): string | null {
+    // Extract base language from locale (e.g., 'en-US' -> 'en', 'es-MX' -> 'es')
+    const match = locale.match(/^([a-z]{2})(?:-[A-Z]{2})?$/);
+    return match ? match[1] : null;
+  }
+
+  /**
+   * Merge base config with localized FTL content
+   */
+  public async mergeConfigs(baseConfig: any, ftlContent: string, clientId: string, entrypoint: string): Promise<any> {
```

Comment:
I don't understand why `any` is necessary here.  We already have types for things we fetch from the CMS...?